### PR TITLE
Prevent fatal error if NULL is provided in array_search under Jetpack Stats

### LINKED ISSUE #50475: Fatal error: `Uncaught TypeError: array_search(): Argument #2 ($haystack) must be of type array, null given`
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

I encountered this fatal while testing on a T51 site that had Woo 9.1.4 installed:

`Uncaught TypeError: array_search(): Argument #2 ($haystack) must be of type array, null given`

Note that this fatal error was already there when I opened up the test site, so I'm not sure how to reproduce it.

### Expected behavior

No fatal error on the site.

### Actual behavior

The fatal error was seen.

### Steps to reproduce

1. Go to the site mentioned in pb22l9-35a-p2#comment-3415.
2. Log in as admin. Reach out to anyone from `@woocommerce/solaris` for credentials.
3. Go to WooCommerce > Status > Logs.
4. Open the log file `fatal-errors-2024-08-06`.
5. Notice that the fatal error mentioned in the description above was repeatedly logged there. 

### WordPress Environment

SSR and the full stacktrace can be found here: pb22l9-35a-p2#comment-3478

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by puntope**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #50475

In this PR we do check if the fields in the jetcpack/stats report data are an array before using them in `array_search` so we prevent a potential fatal

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this PR and also Jetpack
2. Connect WPCOM
3. Go to Analytics -> Reports
4. Visitors and Views shows correctly
5. Request `GET wp-json/wc-analytics/reports/performance-indicators`
6. You get correctly Jetpack/stats/visitors and views.
7. Check WooCommerce -> Status -> Logs 
8. There are NO fatals 

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Admin/API/Reports/PerformanceIndicators/Controller.php [hunk 1]
```diff
@@ -527,8 +527,14 @@ private function get_stats_parts( $full_stat ) {
 	 */
 	public function format_data_value( $data, $stat, $report, $chart, $query_args ) {
 		if ( 'jetpack/stats' === $report ) {
+			$index  = false;
+			$fields = $data['general']->visits->fields;
```
Comment (mikkamp) [inline plugins/woocommerce/src/Admin/API/Reports/PerformanceIndicators/Controller.php]: We can still get three warnings from this line if `$data` isn't formatted correctly. Or worse, another fatal if `$data` is a string. Without looking at the context where this is used, it's marked as a `public` function and the description for the data field is `Data from external endpoint`. Based on that I wouldn't feel confident that it's always going to get passed with the correct format.

**Comment by puntope** [comment]
Howdy @mikkamp I tweaked the code so we do check also if $data['general']->visits->fields is set

Can you take another look?

**Review by mikkamp — APPROVED** [review]
Thanks for the additional changes.

I had a bit of trouble reproducing the fatal error where the data isn't formatted correctly. After connecting Jetpack and viewing the Analytics > Overview I see the Jetpack stats but they don't generate any fatal errors.

image

To mimick the same errors I used the following code snippet:
```php
add_filter(
	'woocommerce_rest_performance_indicators_data_value',
	function ( $data, $stat, $report, $chart, $query_args ) {
		if ( 'jetpack/stats' === $report ) {
			$data['general']->visits->fields = null;
		}
		return $data;
	},
	1,
	5
);
```

Which then did generate the same error as mentioned in the issue:
```
2024-08-16T09:31:40+00:00 Critical Uncaught TypeError: array_search(): Argument #2 ($haystack) must be of type array, null given in /srv/htdocs/wp-content/plugins/woocommerce/src/Admin/API/Reports/PerformanceIndicators/Controller.php:531 
```

It would have been handy to include something like that in the test instructions, otherwise I'm not really confident that we are confirming it's working as intended. After including the fix in this PR as well as the code snippet active I can see that it no longer returns the Jetpack stats and also doesn't generate any fatal errors anymore.

image

I tested the same again by setting `$data` as a string in the code snippet, which produced the same results.

**Comment by puntope** [comment]
> It would have been handy to include something like that in the test instructions, otherwise I'm not really confident that we are confirming it's working as intended. After including the fix in this PR as well as the code snippet active I can see that it no longer returns the Jetpack stats and also doesn't generate any fatal errors anymore.


Right @mikkamp I missed that in the description. What I did was to hardcode the data as string right there in the function. But adding it as a filter also works.

**Pull request merged**